EIPS/eip-1193.md

```diff
@@ -480,15 +482,52 @@ If the accounts available to the Provider change, the Provider **MUST** emit the
 
 The "accounts available to the Provider" change when the return value of `eth_accounts` changes.
 
+## Security Considerations
+
+_This section and its sub-sections are non-normative._
+
+The Provider is intended to pass messages between an Ethereum Client and an Ethereum application.
+It is _not_ responsible for private key or account management; it merely processes RPC messages and emits events.
+Consequently, account security and user privacy need to be implemented in middlewares between the Provider and its Ethereum Client.
+In practice, we call these middleware applications "Wallets," and they usually manage the user's private keys and accounts.
+The Provider can be thought of as an extension of the Wallet, exposed in an untrusted environment, under the control of some third party (e.g. a dapp).
+
+### Handling Adversarial Behavior
+
+Since it is a JavaScript object, consumers can generally perform arbitrary operations on the Provider, and all its properties can be read or overwritten.
+Therefore, it is best to treat the Provider object as though it is controlled by an adversary.
+It is paramount that the Provider implementer protects the user, Wallet, and Client by ensuring that:
+
+- The Provider does not contain any private user data.
+- The Provider and Wallet programs are sandboxed from each other.
```

Comment:
```suggestion
- The Provider and Wallet programs are isolated from each other.
```

Sandboxing is usually a specific heavy-weight type of environmental isolation, but isolation itself is abstract and doesn't specify the safety strategy employed.